typescript/infra/scripts/deploy.ts

```diff
@@ -53,60 +56,94 @@ async function main() {
 
   let config: ChainMap<unknown> = {};
   let deployer: HyperlaneDeployer<any, any>;
-  if (module === Modules.ISM_FACTORY) {
-    config = objMap(envConfig.core, (chain) => true);
-    deployer = new HyperlaneIsmFactoryDeployer(multiProvider);
-  } else if (module === Modules.CORE) {
-    config = envConfig.core;
-    const ismFactory = HyperlaneIsmFactory.fromEnvironment(
-      deployEnvToSdkEnv[environment],
-      multiProvider,
-    );
-    deployer = new HyperlaneCoreDeployer(multiProvider, ismFactory);
-  } else if (module === Modules.INTERCHAIN_GAS_PAYMASTER) {
-    config = envConfig.igp;
-    deployer = new HyperlaneIgpDeployer(multiProvider);
-  } else if (module === Modules.INTERCHAIN_ACCOUNTS) {
-    config = await getRouterConfig(environment, multiProvider);
-    deployer = new InterchainAccountDeployer(multiProvider);
-  } else if (module === Modules.INTERCHAIN_QUERY_SYSTEM) {
-    config = await getRouterConfig(environment, multiProvider);
-    deployer = new InterchainQueryDeployer(multiProvider);
-  } else if (module === Modules.LIQUIDITY_LAYER) {
-    const routerConfig = await getRouterConfig(environment, multiProvider);
-    if (!envConfig.liquidityLayerConfig) {
-      throw new Error(`No liquidity layer config for ${environment}`);
+  switch (module) {
+    case Modules.ISM: {
+      config = envConfig.core;
+      const { ism } = await withIsm(getArgs()).argv;
+      const ismFactory = HyperlaneIsmFactory.fromEnvironment(
+        deployEnvToSdkEnv[environment],
+        multiProvider,
+      );
+      const deployer = new HyperlaneCoreDeployer(multiProvider, ismFactory);
+      // map ism moduletype to an ism config in another file
+      const ismConfig = ismModuleToConfig[ism];
+      await deployer.deployIsms(config, ismConfig);
+
+      return;
+    }
```

Comment:
This is the only contribution in this file, the rest of the changes are just to make it more readable by refactoring to a `switch`